src/smt/term_formula_removal.cpp

```diff
@@ -292,6 +292,12 @@ Node RemoveTermFormulas::runCurrent(std::pair<Node, uint32_t>& curr,
           "a variable introduced due to term-level ITE removal");
       d_skolem_cache.insert(node, skolem);
 
+      // Notice that in very rare cases, two different terms may have the
+      // same purification skolem. An example is the pair of terms
+      // (ite (ite A B C) D E) and (ite k D E), where k is the skolem
+      // introduced to eliminate (ite A B C). For such cases, for simplicity,
+      // we repeat the work of constructing the assertion and proofs below.
```

Comment:
Say why this is not an issue.

Reply:
I think the information I was looking for here is why it's not an issue to have two different terms with the same purification skolem, which is because this will only happen when their original form is the same. As it is one wonders.

Reply:
Ok, I think this documentation doesn't belong here then. I'll add this to `skolem_manager.h`.

Reply:
That's better, thanks!